File: 1065/Main.java
import java.io.IOException;
import java.io.BufferedReader;
import java.io.InputStreamReader;

public class Main {
	public static void main(String [] args) throws IOException {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

		int N = Integer.parseInt(br.readLine());

		int count = 0 ;

		for (int i = 1 ; i <= N ; i++) count += hansu(i);

		System.out.print(count);
	}

	public static int hansu(Integer num) {
		if (num >= 1 && num <= 99) return 1;
		
		String str = num.toString();

		char [] chars = str.toCharArray();	

		int len = chars.length;

		int [] arr = new int[len];

		for (int i = 0 ; i < len; i++) {
			arr[i] = chars[i] - '0';
		}

		int diff  = arr[0] - arr[1];
		int diff2 = arr[1] - arr[2];

		if (len == 3 && diff == diff2) {
			return 1;
		}

		return 0;
	}
}

File: 1065/main.py
def is_han(number):
    if number <= 99:
        return True

    string = str(number)
    diff = int(string[0]) - int(string[1])
    length = len(string)
    if length == 3:
        diff2 = int(string[1]) - int(string[2])
        if diff2 == diff:
            return True
        else:
            return False

    if length == 4:
        diff2 = int(string[1]) - int(string[2])
        diff3 = int(string[2]) - int(string[3])
        if diff2 == diff and diff3 == diff and diff2 == diff3:
            return True
        else:
            return False


n = int(input())
count = 0

for idx in range(1, n+1):
    if is_han(idx) is True:
        count += 1


print(count)

